src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -163,32 +163,98 @@ public void run() {
     return OutputView.PrintView.newBuilder().format(specs.get(0)).args(args).build();
   }
 
-  private OutputView.LoopView loopView(
+  private OutputView loopView(
       OutputSpec.LoopStatement loop,
       MethodContext context,
       SampleValueSet valueSet,
       ScopeTable localVars,
       CallingForm form) {
+    if (!loop.getCollection().isEmpty() && loop.getMap().isEmpty()) {
+      Preconditions.checkArgument(
+          !loop.getVariable().isEmpty(),
+          "Bad format: `variable` must be specified if `collection` is specified.");
+      Preconditions.checkArgument(
+          loop.getKey().isEmpty() && loop.getValue().isEmpty(),
+          "Bad format: `key` and `value` can't be specified if `collection is specified.");
```

Comment:
nit: "... neither `key` nor `value` can be ..."
nit: end quote after `collection`